services/horizon/internal/db2/history/trade_aggregation.go

```diff
@@ -123,51 +123,101 @@ func (q *TradeAggregationsQ) WithEndTime(endTime strtime.Millis) (*TradeAggregat
 	}
 }
 
-// GetSql generates a sql statement to aggregate Trades based on given parameters
-func (q *TradeAggregationsQ) GetSql() sq.SelectBuilder {
-	var orderPreserved bool
-	orderPreserved, q.baseAssetID, q.counterAssetID = getCanonicalAssetOrder(q.baseAssetID, q.counterAssetID)
-
-	var bucketSQL sq.SelectBuilder
+func (q *TradeAggregationsQ) getRawTradesSql(orderPreserved bool) sq.SelectBuilder {
+	var rawTradesSQL sq.SelectBuilder
 	if orderPreserved {
-		bucketSQL = bucketTrades(q.resolution, q.offset)
+		rawTradesSQL = bucketTrades(q.resolution, q.offset)
 	} else {
-		bucketSQL = reverseBucketTrades(q.resolution, q.offset)
+		rawTradesSQL = reverseBucketTrades(q.resolution, q.offset)
 	}
 
-	bucketSQL = bucketSQL.From("history_trades_60000").
+	rawTradesSQL = rawTradesSQL.Join("timestamp_range r ON 1=1")
+	rawTradesSQL = rawTradesSQL.From("history_trades_60000 AS tr").
 		Where(sq.Eq{"base_asset_id": q.baseAssetID, "counter_asset_id": q.counterAssetID})
 
 	//adjust time range and apply time filters
-	bucketSQL = bucketSQL.Where(sq.GtOrEq{"timestamp": q.startTime})
-	if !q.endTime.IsNil() {
-		bucketSQL = bucketSQL.Where(sq.Lt{"timestamp": q.endTime})
-	}
+	rawTradesSQL = rawTradesSQL.Where(
+		fmt.Sprintf("r.max_ts >= %s", formatBucketTimestamp(q.resolution, q.offset, "tr")))
+	rawTradesSQL = rawTradesSQL.Where(
+		fmt.Sprintf("r.min_ts <= %s", formatBucketTimestamp(q.resolution, q.offset, "tr")))
 
 	if q.resolution != 60000 {
 		//ensure open/close order for cases when multiple trades occur in the same ledger
-		bucketSQL = bucketSQL.OrderBy("timestamp ASC", "open_ledger_toid ASC")
+		rawTradesSQL = rawTradesSQL.OrderBy("timestamp ASC", "open_ledger_toid ASC")
 		// Do on-the-fly aggregation for higher resolutions.
-		bucketSQL = aggregate(bucketSQL)
 	}
+	return rawTradesSQL
+}
 
-	return bucketSQL.
+// GetSql generates a sql statement to aggregate Trades based on given parameters
+func (q *TradeAggregationsQ) GetSql() sq.SelectBuilder {
+	var orderPreserved bool
+	orderPreserved, q.baseAssetID, q.counterAssetID = getCanonicalAssetOrder(q.baseAssetID, q.counterAssetID)
+
+	var bucketSQL sq.SelectBuilder
+	bucketSQL = aggregate("raw_trades")
+	bucketSQL = bucketSQL.
 		Limit(q.pagingParams.Limit).
 		OrderBy("timestamp " + q.pagingParams.Order)
+
+	bucketSQL = bucketSQL.Prefix("WITH last_200_ts AS (?),",
+		last200Ts(q.baseAssetID, q.counterAssetID, q.resolution, q.offset, q.startTime, q.endTime))
+	bucketSQL = bucketSQL.Prefix("timestamp_range AS (?),",
+		timestampRange())
+	bucketSQL = bucketSQL.Prefix("raw_trades AS (?)",
+		q.getRawTradesSql(orderPreserved))
+
+	return bucketSQL
```

Comment:
bit of a nit, but I find the existing
```go
query = sq.Select(...).
	From(...).
	Where(...)
```
syntax to be a more readable (applies to `last200Ts`, below, as well).

e.g.
```suggestion
	return bucketSQL.
		Limit(q.pagingParams.Limit).
		OrderBy("timestamp " + q.pagingParams.Order).
		Prefix("WITH last_200_ts as (?),", 
			last200Ts(q.baseAssetID, q.counterAssetID, q.resolution, q.offset, q.startTime, q.endTime)).
		Prefix("timestamp_range as (?),", timestampRange()).
		Prefix("raw_trades as (?)", q.getRawTradesSql(orderPreserved))
```

Can we keep that consistent throughout this change?

Reply:
Sure, I've made changes where appropriate to use this syntax.